### ISSUE TITLE
Default coordinate system for layers, and project/unproject

### ISSUE THREAD

**Issue by ralphstodomingo** [uber/deck.gl#2278]
#### Background 
First off, congratulations on the 6.1.0 release!
As detailed in #936 and #818, I have tried before to make a GeoJSON editor within deck.gl, but had hit issues due to precision on a Windows development machine.
Seeing that there was a new coordinate system released in 6.1.0, I went over the code I had dating back to the 5.0 release and adapted it to 6.1.

Initially, this was the result, which was the original problem outlined in #936:
![precision-issues](https://user-images.githubusercontent.com/15341353/45541480-db834b80-b841-11e8-901f-64ce5fc69199.gif)
I also tried setting `fp64: true` but still, I encountered that long startup for the layer that involved the browser lagging beyond usability.
After the initial disappointment that this feature I'm making might still not be feasible in deck.gl, I wondered whether this statement on the docs were true:

> High-Precision Geospatial Projection (Experimental)
> The projection algorithm used for geospatial coordinates (layers with coordinateSystem: COORDINATE_SYSTEM.LNGLAT) has supplemented with a "hybrid" projection/offset based implementation (COORDINATE_SYSTEM.LNGLAT_EXPERIMENTAL) that rivals 64-bit precision at 32-bit speeds. This mode is expected to make the use of fp64 precision unnecessary for most applications, which in turn should increase application performance and avoid issues on untested graphics drivers.

There wasn't anything in the docs that mentioned where to export these coordinate systems or set them in layers, so I looked around in the source code until I found [this constants file](https://github.com/uber/deck.gl/blob/master/modules/core/src/shaderlib/project/constants.js) and tried to set my layer's coordinate system manually to:
```
 new GeoJsonLayer({
  id: 'geojson-layer-feature',
  data: drawFeatureCache,
  ...
  coordinateSystem: 4,
}),
```

And here's the result:
![precision-issues-fixed](https://user-images.githubusercontent.com/15341353/45541785-8a278c00-b842-11e8-991b-06f39cde1155.gif)
🙌👏🎉 Thank you for this new coordinate system!

A few questions and suggestions on my end:
1. Isn't this new coordinate system (`LNGLAT_AUTO_OFFSET`) not set as default for the layers that support it?
2. Should the docs be updated as to how these coordinate systems can be set in the layers?
3. I'm still using this for my screen pixel -> coordinate conversion:
```
let lngLat = deckGLInstance.deck.layerManager.context.viewport.unproject([
  event.clientX,
  event.clientY,
]);
```
Right now, I only have a component for this. Is this something that can be converted to a custom layer?
Would be definitely interested if there was a different way to get these coordinates, as this looks hacky at best.

I'd be willing to send  a PR updating the docs if this looks good to you!
Again, thanks for all your hard work!

#### To Do List 
  - [ ] Add label and assign to milestone 
  - [ ] Coding
  - [ ] Doc update
  - [ ] Whats new update 
  - [ ] Test

**Comment by ibgreen** [comment]
> I'm still using this for my screen pixel -> coordinate conversion:

Yes, I seem to remember we talked about this quite some time ago. This one hasn't been addressed. I think the issue is that while adding a wrapper for the geospatial single view case is easy, doing it right while handling all our coordinate systems and also multi-viewports takes some though.

FWIW, there are number of issues open on improving the projection system, and an RFC has been started.

https://github.com/uber/deck.gl/issues/2151

**Comment by ibgreen** [comment]
Thanks for including the GIFs, I really like your editing capabilities. 

Have you looked at [nebula.gl](https://neb.gl/#/)? It is an editing framework for deck.gl. 

You may want to head over there and connect with @georgios-uber, perhaps some collaboration or exchange of ideas is possible?
